scaler.rb

```diff
@@ -36,28 +34,82 @@
   },
 }
 
-updates = {
+config_updates = {}
+
+formation_updates = {
   "process" => options[:process],
   "quantity" => options[:quantity]
 }
 
-config_updates = {}
+preboot_enabled   = heroku.app_feature.info(options[:app], 'preboot')["enabled"]
+current_formation = heroku.formation.info(options[:app], options[:process])
+current_config    = heroku.config_var.info(options[:app])
 
-if options[:size]
-  options[:size].upcase!
-  updates["size"] = options[:size]
-  if default_config[options[:size]]['RUBY_GC_HEAP_GROWTH_MAX_SLOTS']
-    config_updates['RUBY_GC_HEAP_GROWTH_MAX_SLOTS'] = default_config[options[:size]]['RUBY_GC_HEAP_GROWTH_MAX_SLOTS']
-  else
-    config_updates['RUBY_GC_HEAP_GROWTH_MAX_SLOTS'] = nil
+# Default to updating config first, then formation
+commands = [:config, :formation]
+
+if options[:web_concurrency]
+  config_updates['WEB_CONCURRENCY'] = options[:web_concurrency]
+end
+
+# Are we changing dyno sizes?
+if options[:size] && options[:size] != current_size
+  formation_updates["size"] = options[:size]
+  # setting values to nil will unset the config value
+  config_updates['RUBY_GC_HEAP_GROWTH_MAX_SLOTS'] = default_config[options[:size]]['RUBY_GC_HEAP_GROWTH_MAX_SLOTS']
+  # only set WEB_CONCURRENCY to the default if we did not set it as an option
+  config_updates['WEB_CONCURRENCY'] ||= default_config[options[:size]]['WEB_CONCURRENCY']
```

Comment:
these config updates aren't necessary if these values are already set... so maybe we should compare them to existing values before setting them. That way we don't have to sleep if no config changes are made before scaling to 1 dyno.